test-libz-rs-sys/src/gz.rs

```diff
@@ -6,42 +6,83 @@ use std::ffi::CString;
 use std::ptr;
 
 // Generate a file path relative to the project's root
-macro_rules! path {
+macro_rules! crate_path {
     ($str:expr) => {
         concat!(env!("CARGO_MANIFEST_DIR"), "/", $str)
     };
 }
 
-unsafe fn test_open(path: &str, mode: &str, should_succeed: bool) {
-    let cpath = CString::new(path).unwrap();
-    let cmode = CString::new(mode).unwrap();
-    let handle = gzopen(cpath.as_ptr(), cmode.as_ptr());
-    assert_eq!(should_succeed, !handle.is_null(), "{path} {mode}");
-    if !handle.is_null() {
-        assert_eq!(gzclose(handle), Z_OK);
-    }
+// Generate a file path relative to a specified directory prefix
+macro_rules! path {
+    ($prefix:expr, $str:expr) => {{
+        let mut path_buf = $prefix.to_path_buf();
+        path_buf.push($str);
+        path_buf.as_path().to_str().unwrap().to_owned()
+    }};
+}
```

Comment:
is there a particular reason for the macros? Usually standard functions have better editor support and are easier to reason about.

Reply:
I made `test_open` a macro so failures in its `assert` calls will be reported by the test driver as happening on the line number where the macro is called. But the other macros can easily be functions. I'll push an update to this PR later today to make them functions.